06-reactJs/components/new-movie-form.js

```diff
@@ -0,0 +1,31 @@
+var React = require('react');
+var Form = require('./form');
+var MovieList = require('./movie-list');
+
+var NewMovieForm = React.createClass({
+    
+    getInitialState: function () {
+        return {
+            data: [
+                {title: 'Fight club', genre: 'Thriller', favorite: true},
+                {title: 'Warrior', genre: 'Action', favorite: true}
+            ]
+        };
+    },
+
+    render: function () {
+        return (
+            <div className='NewMovieForm'>
+                <Form onMovieSubmit={this.handleOnMovieSubmit}></Form>
```

Comment:
I think the idea behind these two components (*new-movie-form.js* and *form.js*) is to have a generic ``Form`` component and reuse it where you need, right?
If that's the case, I like the idea, but have in mind that like it is right now, it's not really decoupled. Everything inside your ``Form`` component makes reference to movies (your states, callbacks, etc.).
For this case, I would skip this file and handle everything in your *form.js* file (you can rename it to *movie-form.js* or something like that).
Hope I'm being clear enough, if that's not the case, reach out to me!